### PR TITLE
Added Thrift server for saivs target

### PR DESCRIPTION
**Pull request by taras-keryk-plv**
Added Thrift server for saivs target in standalone mode.
 Build target: 
./build.sh -a BCM56850 -t saivs -s thrift
Run target:
./run.sh -a BCM56850 -t saivs -s thrift -c start
Run thrift test example:
./exec.sh -a BCM56850 -t saivs pytest --testbed=saivs_standalone tests -k "access_to_access" -v
```
taras.keryk@pcn-tkeryk-u20:~/SAI-Challenger$ ./exec.sh -a BCM56850 -t saivs pytest --testbed=saivs_thrift_standalone tests -k "access_to_access" -v
===================================================== test session starts =====================================================
platform linux -- Python 3.7.3, pytest-7.2.1, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.7.3', 'Platform': 'Linux-5.15.0-60-generic-x86_64-with-debian-10.13', 'Packages': {'pytest': '7.2.1', 'pluggy': '1.0.0'}, 'Plugins': {'dependency': '0.5.1', 'metadata': '2.0.4', 'html': '3.2.0'}}
rootdir: /sai-challenger, configfile: pytest.ini
plugins: dependency-0.5.1, metadata-2.0.4, html-3.2.0
collected 560 items / 559 deselected / 1 selected                                                                             

tests/test_l2_basic.py::test_l2_access_to_access_vlan PASSED                                                            [100%]

============================================== 1 passed, 559 deselected in 0.24s ==============================================

===========================================
     SAI Challenger run options
===========================================

 Docker image type  : standalone
 ASIC name          : BCM56850
 ASIC target        : saivs
 Platform path      : ./npu/broadcom/BCM56850
 Container name     : sc-bcm56850-saivs-run
 Command            : pytest --testbed=saivs_thrift_standalone tests -k access_to_access -v

===========================================




```

### CODE REVIEW

File: Dockerfile.saithrift [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        && dpkg -i libsaivs-dev_1.0.0_amd64.deb \
+        && dpkg -i libsaimetadata_1.0.0_amd64.deb \
+        && dpkg -i libsaimetadata-dev_1.0.0_amd64.deb
+RUN cp 0001-improve-enum-values-integration-check-1727.patch sonic-sairedis \
+        && cd sonic-sairedis && patch -p1 < 0001-improve-enum-values-integration-check-1727.patch \
+        && cd SAI && make saithrift-install \
+        && cp meta/saimetadatautils.c /sai/gen_attr_list/ \
+        && cp meta/saimetadata.c /sai/gen_attr_list/ \
+        && cp meta/saiserialize.c /sai/gen_attr_list/ 
+        #&& cd - \
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:144]: please remove if not needed

File: npu/broadcom/BCM56850/saivs/configs/supervisord.conf.saithrift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+[program:redis]
+command=/usr/bin/redis-server --unixsocket /var/run/redis/redis.sock
+priority=2
+autostart=true
+autorestart=true
+stdout_logfile=syslog
+stderr_logfile=syslog
+dependent_startup_wait_for=rsyslogd:running
+
+
```
Comment (andriy-kokhan) [inline npu/broadcom/BCM56850/saivs/configs/supervisord.conf.saithrift:41]: please remove one empty line

File: run.sh [hunk 3]
```diff
@@ -8,6 +8,7 @@ trap 'last_command=$current_command; current_command=$BASH_COMMAND' DEBUG
 # echo an error message before exiting
 trap 'echo ERROR: "\"${last_command}\" command filed with exit code $?."' ERR
 
+#SAI_INTERFACE="redis"
```
Comment (andriy-kokhan) [inline run.sh:11]: please clean-up if not needed

File: run.sh [hunk 4]
```diff
@@ -67,6 +68,10 @@ while [[ $# -gt 0 ]]; do
             COMMAND="$2"
             shift
         ;;
+        #"-s"|"--sai_interface")
```
Comment (andriy-kokhan) [inline run.sh:71]: please clean-up if not needed

File: testbeds/saivs_standalone.json [hunk 5]
```diff
@@ -7,10 +7,10 @@
     "target": "saivs",
     "sku": null,
     "client": {
-      "type": "redis",
+      "type": "thrift",
```
Comment (andriy-kokhan) [inline testbeds/saivs_standalone.json:10]: lets have a separate file for Thrift case

File: npu/broadcom/sai_npu.py [hunk 6]
```diff
@@ -12,8 +12,8 @@ def reset(self):
         attr = [
             "SAI_SWITCH_ATTR_SRC_MAC_ADDRESS",      "52:54:00:EE:BB:70",
             "SAI_SWITCH_ATTR_FDB_AGING_TIME",       "600",
-            "SAI_SWITCH_ATTR_TPID_INNER_VLAN",      "33024",    # 0x8100
-            "SAI_SWITCH_ATTR_TPID_OUTER_VLAN",      "34984",    # 0x88A8
+            #"SAI_SWITCH_ATTR_TPID_INNER_VLAN",      "33024",    # 0x8100
```
Comment (andriy-kokhan) [inline npu/broadcom/sai_npu.py:15]: please provide some comment why do we commented this out

**Review by andriy-kokhan — APPROVED** [review]

**Comment by anton7811** [comment]
@taras-keryk-plv @andriy-kokhan to run this project I had to remove all previous sai-c images, remove other my projects, and finally it anyway used all free space on the device and test failed.

**Comment by andriy-kokhan** [comment]
> @taras-keryk-plv @andriy-kokhan to run this project I had to remove all previous sai-c images, remove other my projects, and finally it anyway used all free space on the device and test failed.

@anton7811 , thank you for your input. @taras-keryk-plv , please check and fix by next PR.

**Pull request merged**